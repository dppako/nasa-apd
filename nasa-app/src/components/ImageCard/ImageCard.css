.ImageCard {
  padding: 10px; 
}

.ImageCard__image {
  max-width: 100%;
}  