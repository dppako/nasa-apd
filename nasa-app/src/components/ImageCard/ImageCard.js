import React from 'react';
import PropTypes from 'prop-types';
import './ImageCard.css';

const ImageCard = ({ imageData = {} }) => {
  const { title, url, explanation, date, copyright } = imageData;
  return (
    <article className='ImageCard'>
      <header>
        <h2 className='siimple-h3'>{title}</h2>
      </header>
      <img className='ImageCard__image' src={url} alt={title} />
      <p>
        <small className='siimple-small'>{copyright}, {date}</small>
      </p>
      <p className='siimple-p'>{explanation}</p>
    </article>
  );
};

ImageCard.propTypes = {
  imageData: PropTypes.object
};

export default ImageCard;