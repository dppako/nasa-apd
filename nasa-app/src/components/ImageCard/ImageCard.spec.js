import React from 'react';
import renderer from 'react-test-renderer';
import ImageCard from './ImageCard';

describe('ImageCard Component', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <ImageCard />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Given imageData', () => {
    const imageData = {
      title: 'test title',
      url: 'test url',
      explanation: 'test explanation',
      date: 'test date',
      copyright: 'test copyright'
    };

    it('matches snapshot', () => {
      const component = renderer.create(
        <ImageCard imageData={imageData} />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});