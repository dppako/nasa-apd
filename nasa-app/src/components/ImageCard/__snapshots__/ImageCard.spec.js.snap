// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard Component Given imageData matches snapshot 1`] = `
<article
  className="ImageCard"
>
  <header>
    <h2
      className="siimple-h3"
    >
      test title
    </h2>
  </header>
  <img
    alt="test title"
    className="ImageCard__image"
    src="test url"
  />
  <p>
    <small
      className="siimple-small"
    >
      test copyright
      , 
      test date
    </small>
  </p>
  <p
    className="siimple-p"
  >
    test explanation
  </p>
</article>
`;

exports[`ImageCard Component matches snapshot 1`] = `
<article
  className="ImageCard"
>
  <header>
    <h2
      className="siimple-h3"
    />
  </header>
  <img
    alt={undefined}
    className="ImageCard__image"
    src={undefined}
  />
  <p>
    <small
      className="siimple-small"
    >
      , 
    </small>
  </p>
  <p
    className="siimple-p"
  />
</article>
`;
